src/components/Missions/Missions.js

```diff
@@ -1,7 +1,47 @@
-import React from 'react';
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { getMissions, reserve } from '../../redux/missions/missionSlice';
 
-const Missions = () => (
-  <div>Missions</div>
-);
+const Missions = () => {
+  const { missions, isLoading, error } = useSelector((state) => state.missions);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getMissions());
+  }, [dispatch]);
+
+  return (
+    <div className="missions">
+      {isLoading && <div>Loading...</div>}
+      {error && <div>Sorry, something went wrong.</div>}
+      {missions && (
+        <table>
+          <thead>
+            <tr>
+              <th>Mission</th>
+              <th>Description</th>
+              <th>Status</th>
+              <th>{' '}</th>
+            </tr>
+          </thead>
+          <tbody>
+            {missions.map((mission) => (
+              <tr key={mission.mission_id}>
+                <td>{mission.mission_name}</td>
+                <td>{mission.description}</td>
+                <td>{mission.reserved ? 'Active Member' : 'NOT A MEMBER'}</td>
+                <td>
+                  <button type="button" onClick={() => dispatch(reserve(mission.mission_id))}>
+                    {mission.reserved ? 'Leave Mission' : 'Join Mission'}
+                  </button>
+                </td>
+              </tr>
+            ))}
+          </tbody>
+        </table>
+      )}
+    </div>
+  );
+};
```

Comment:
refactored code using components
By breaking down the original code into smaller components, you improve the readability, reusability, and maintainability of your codebase. It also follows a more modular approach, making it easier to reason about the different parts of your application.

Reply:
Hi @ahmidou09 thanks for your advice, but I have some questions and doubts about your requirements:
Making mor modular and divide into components makes me repeat code because I have to import two times the same hooks for different files:

![image](https://github.com/victormc13/Space-Travelers-Hub/assets/110580299/33fe9970-4c28-40ae-ba3c-c9f25d8f7ec4)

there I'm importing the { useSelector, useDispatch } two times, one for Missions component and another for Table component:

![image](https://github.com/victormc13/Space-Travelers-Hub/assets/110580299/065344a6-7421-415e-a9fc-69cf65c2bbcb)

Some advice for this? thanks in advance.